airflow-core/src/airflow/models/taskinstance.py

```diff
@@ -285,7 +285,12 @@ def clear_task_instances(
             # the task is terminated and becomes eligible for retry.
             ti.state = TaskInstanceState.RESTARTING
         else:
-            ti_dag = dag if dag and dag.dag_id == ti.dag_id else dag_bag.get_dag(ti.dag_id, session=session)
+            dr = ti.dag_run
```

Comment:
Is it guaranteed that ti.dag_run cannot be none?

Reply:
i believe this guarantees it?
<img width="498" alt="image" src="https://github.com/user-attachments/assets/f5d7b27c-ba3a-4431-ad22-39c221bffac5" />

Reply:
Ok. I think here: https://github.com/apache/airflow/blob/468bfe339ed3d5053db2a9776b6f9bbf4f7a04de/airflow-core/src/airflow/models/dag.py#L1048